neo/Consensus/ConsensusService.cs

```diff
@@ -534,11 +542,20 @@ public static Props Props(IActorRef localNode, IActorRef taskManager, Store stor
         private void RequestChangeView()
         {
             if (context.WatchOnly()) return;
-            byte expectedView = context.ChangeViewPayloads[context.MyIndex]?.GetDeserializedMessage<ChangeView>().NewViewNumber ?? 0;
-            if (expectedView < context.ViewNumber) expectedView = context.ViewNumber;
+            // Request for next view is always one view more than the current context.ViewNumber
+            // Nodes will not contribute for changing to a view higher than (context.ViewNumber+1), unless they are recovered
+            // The latter may happen by nodes in higher views with, at least, `M` proofs
+            byte expectedView = context.ViewNumber;
             expectedView++;
-            Log($"request change view: height={context.BlockIndex} view={context.ViewNumber} nv={expectedView}");
             ChangeTimer(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (expectedView + 1)));
+            if ((context.CountCommitted() + context.CountFailed()) > context.F())
+            {
+                Log($"Not requesting change view to nv={expectedView} because nc={context.CountCommitted()} nf={context.CountFailed()}");
+                // asking for recovery (ping)
+                ConsensusNodePingForRecovery();
```

Comment:
It should probably send it’s request using its current view number instead of view 0.

Reply:
Pushed a commit to address this.